CMakeLists.txt

```diff
@@ -187,13 +187,13 @@ if(WITH_CUDA)
       message(STATUS "Found HIP compiler: ${HIP_HIPCC_EXECUTABLE}")
       set(CUDA 1)
       set(HIP 1)
-      list(APPEND HIP_HCC_FLAGS "-I${HIP_ROOT_DIR}/include -Wno-c99-designator -Wno-macro-redefined -Wno-duplicate-decl-specifier -std=c++14")
+      list(APPEND HIP_HCC_FLAGS "-I${HIP_ROOT_DIR}/include -I${HIP_ROOT_DIR}/../include -Wno-c99-designator -Wno-macro-redefined -Wno-duplicate-decl-specifier -std=c++14")
```

Comment:
`HIP_HCC_FLAGS` expands to `"-I/opt/rocm/include -I/opt/rocm/../include -Wno-c99..."` on ROCm 3.0, but `/opt/rocm/../include` doesn't exist. Is there a way in CMake to find the path of the parent ROCm folder? This would allow us to write `"-I${HIP_ROOT_DIR}/include -I${ROCM_ROOT_DIR}/include -Wno-c99..."` and save us some headache when debugging this CMake logic.

Reply:
Unfortunately there is not. So maybe we should merge your patch instead of this pull request that contains an explicit version check.

Reply:
We could do like in [pytorch/pytorch:torch/utils/cpp_extension.py#L53-L73](https://github.com/pytorch/pytorch/blob/d30fa4837e10505d172e0cc7a6849b1a07aeb573/torch/utils/cpp_extension.py#L53-L73), e.g.
```cmake
      string(REGEX REPLACE "^(.+)/bin/hipcc$" "\\1" ROCM_HOME ${HIP_HIPCC_EXECUTABLE})
      list(APPEND HIP_HCC_FLAGS "-I${HIP_ROOT_DIR}/include -I${ROCM_HOME}/include -Wno-c99-designator -Wno-macro-redefined -Wno-duplicate-decl-specifier -std=c++14")
```
But this seems superfluous given that we already hardcode the path to `/opt/rocm`:
https://github.com/espressomd/espresso/blob/a4596e3e07e3c948a52217a32ced0e962f0bf165/CMakeLists.txt#L182

Why not directly do the following?
```cmake
set(ROCM_HOME "/opt/rocm")
list(APPEND CMAKE_MODULE_PATH "${ROCM_HOME}/hip/cmake") 
# ...
list(APPEND HIP_HCC_FLAGS "-I${HIP_ROOT_DIR}/include -I${ROCM_HOME}/include -Wno-c99-designator -Wno-macro-redefined -Wno-duplicate-decl-specifier -std=c++14")
```

Reply:
`ROCM_HOME` sounds like a sensible solution.